com.ibm.streamsx.topology/opt/python/packages/streamsx/scripts/sc.py

```diff
@@ -51,45 +52,95 @@ def _parse_dependencies():
-            local_toolkits.append(tk)
+            local_toolkits_paths.append(x)
         else:
             # directory consisting of toolkits, get list and check which are local_toolkits, add them
             sub_directories = glob(x + "*/")
             for y in sub_directories:
                 if _is_local_toolkit(y):
-                    tk = _get_toolkit(y)
-                    local_toolkits.append(tk)
+                    local_toolkits_paths.append(y)
+
+    # For each direct path to a toolkit, parse the name & version, and convert it to a _LocalToolkit object
+    for tk_path in local_toolkits_paths:
+        # Get the name & version of the toolkit that is in the directory tk_path
+        root = ET.parse(tk_path + "/info.xml").getroot()
+        identity = root.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}identity')
+        name = identity.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}name')
+        version = identity.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}version')
+
+        toolkit_name = name.text
+        toolkit_version = version.text
+
+        tk = _LocalToolkit(toolkit_name, toolkit_version, tk_path)
+        local_toolkits.append(tk)
+
+    return local_toolkits
 
-    # Once we have all local toolkits, check if they correspond to any dependencies that we have, if so, add them
-    for toolkit in local_toolkits:
-        if toolkit.name in dependencies:
-            _check_satisfies_version(toolkit, dependencies[toolkit.name], topo)
+def _get_local_toolkit(dependency_name, dependency_version, local_toolkits):
+    """ Helper function that given a required dependency, and a list of local toolkits, finds and returns the local toolkit of the same name,
+    if it satisfies the required dependency version, else returns None.
 
-def _get_toolkit(path):
-    # Get the name & version of the toolkit that is in the directory PATH
-    root = ET.parse(path + "/info.xml").getroot()
-    identity = root.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}identity')
-    name = identity.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}name')
-    version = identity.find('{http://www.ibm.com/xmlns/prod/streams/spl/toolkitInfo}version')
+    Arguments:
+        dependency_name {String} -- The name of the required dependency
+        dependency_version {String} -- The version # or version range of the required dependency
+        local_toolkits {List} -- A list of _LocalToolkit objects, giving all local toolkits
 
-    toolkit_name = name.text
-    toolkit_version = version.text
-    return _LocalToolkit(toolkit_name, toolkit_version, path)
+    Returns:
+        _LocalToolkit -- The matching local toolkit, or None if it doesn't exist
+    """
+    # Get all local toolkits that have the same name as our dependency
+    temp = [x for x in local_toolkits if x.name == dependency_name]
+    # For each toolkit that matches the name
+    for x in temp:
```

Comment:
As a style item `for x in temp` is hard for others to read as it's not clear what `x` or `temp` is. Useful to have variables named for what they represent.